prov/gni/src/gnix_msg.c

```diff
@@ -1044,6 +1050,8 @@ static int __smsg_rndzv_fin(void *data, void *msg)
 				"GNI_SmsgRelease returned %s\n",
 				gni_err_str[status]);
 		ret = gnixu_to_fi_errno(status);
+	} else {
+		ret = gnixu_to_fi_errno(hdr->status);
```

Comment:
I see your point now. Should we be counting failures? Do I even need to return the error code to __smsg_rndzv_fin?